### PR TITLE
SDI-71: 📝 Add more dtos and convert model classes to latest swagger

### PR DESCRIPTION
**Pull request by petergphillips**
- SDI-71: 📝 Add more dtos and convert model classes to latest swagger
- SDI-71: 📝 Add more dtos and convert model classes to latest swagger

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/api/model/StaffLocationRoleDto.kt [hunk 1]
```diff
@@ -3,24 +3,24 @@ import java.math.BigDecimal
...
-  var fromDate: LocalDate? = null,
-  var toDate: LocalDate? = null,
-  var position: String? = null,
-  var positionDescription: String? = null,
-  var role: String? = null,
-  var roleDescription: String? = null,
-  var scheduleType: String? = null,
-  var scheduleTypeDescription: String? = null,
-  var hoursPerWeek: BigDecimal? = null,
+  val staffId: Long?,
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/api/model/StaffLocationRoleDto.kt:6]: can now make these `val` as using our own new mapper instead

File: src/main/java/uk/gov/justice/hmpps/prison/repository/mapping/DataClassByColumnRowMapper.kt [hunk 2]
```diff
@@ -21,6 +21,9 @@ class DataClassByColumnRowMapper<T>(mappedClass: Class<T>) : BeanPropertyRowMapp
   private val mappedConstructor: Constructor<T>
   private val constructorParametersAsColumnNames: List<String>
   private val constructorParameterTypes: List<TypeDescriptor>
+  // fieldMap is used for building up sorting, mapping the java parameter name to the database column name
+  // for example so can then pass "lastName,startTime" and will then sort by those fields
+  val fieldMap: Map<String, FieldMapper>
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/mapping/DataClassByColumnRowMapper.kt:26]: change here to add new `fieldMap` which is used by some of the repositories

File: src/main/java/uk/gov/justice/hmpps/prison/repository/mapping/DataClassByColumnRowMapper.kt [hunk 3]
```diff
@@ -69,4 +76,11 @@ class DataClassByColumnRowMapper<T>(mappedClass: Class<T>) : BeanPropertyRowMapp
     }
     return BeanUtils.instantiateClass(mappedConstructor, *args)
   }
+
+  override fun getColumnValue(rs: ResultSet, index: Int, paramType: Class<*>): Any? {
+    return when (paramType) {
+      Boolean::class.javaObjectType -> "Y" == rs.getObject(index) || rs.getBoolean(index)
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/mapping/DataClassByColumnRowMapper.kt:82]: note the `.javaObjectType` which translates to `java.lang.Boolean` so we have the same functionality as in the [StandardBean](https://github.com/ministryofjustice/prison-api/blob/1e1ecef6c358db4fd5af4b6d456484f9e725716f/src/main/java/uk/gov/justice/hmpps/prison/repository/mapping/StandardBeanPropertyRowMapper.java#L71) version.  Didn't need the local date (time) versions as spring translates these fine these days

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**